app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class SessionsController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by email: params.dig(:session, :email)&.downcase
```

Comment:
dig là gì, vì sao dùng dig

Reply:
dig giúp truy cập vào các giá trị của Hash hoặc Array một cách an toàn và dễ dàng.
Dig của em sẽ có dạng Parameters {"authenticity_token"=>"wSF9DVloHLH5ca9gvHli940gORPj1qP880jqtRQqMVfulNCtX3xlhq4hvY2XEzPn6j6ChWoMLi1R7G0lzG_Mzw", "session"=>{"email"=>"user02@gmail.com", "password"=>"1"}, "commit"=>"Log in", "controller"=>"sessions", "action"=>"create"} permitted: false>

Reply:
để thân thiện người dùng hơn
em nên tạo ra method load_user để handle code `user = User.find_by email: params.dig(:session, :email)&.downcase` và đồng thời xử lý khi user nil trong mehtod load_user để có mess cho chính xác hơn

còn ở method create thì em chỉ cần all `authenticate` và cũng ko cần `&.` làm gì, vì vào tới đây thì sure là user đã khác nil ròi hì

dùng before_action :load_user, only: :create